pandas/_libs/src/vendored/ujson/python/objToJSON.c

```diff
@@ -1538,13 +1550,24 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
         tc->type = JT_UTF8;
         return;
     } else if (PyArray_IsScalar(obj, Datetime)) {
+        npy_int64 longVal;
         if (((PyDatetimeScalarObject *)obj)->obval == get_nat()) {
             tc->type = JT_NULL;
             return;
         }
+        PyArray_Descr *dtype = PyArray_DescrFromScalar(obj);
+        if (dtype->type_num == NPY_OBJECT) {
+            PyErr_Format(PyExc_ValueError, "Could not get resolution of datetime");
```

Comment:
Any reason we aren't returning here? Feels a bit off to let execution continue after setting error

Reply:
My bad, updated.